### PR TITLE
H*(RP2\/S1, Z/2)

### PR DESCRIPTION
**Pull request by thomas-lamiaux**

### CODE REVIEW

File: Cubical/Cohomology/EilenbergMacLane/Rings/Z2-properties.agda [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  ; ·IdR      to ·ℤ/2IdR
+  ; ·DistR+   to ·ℤ/2DistR+
+  ; ·Comm     to ·ℤ/2Comm
+  ; is-set    to isSetℤ/2     )
+
+module _
+  {X : Type ℓ-zero}
+  where
+
+  -- This notation is needed as G'' shouldn't have been put implicit
```
Comment (aljungstrom) [inline Cubical/Cohomology/EilenbergMacLane/Rings/Z2-properties.agda:66]: This is a bit strong maybe?;) It's not needed when working over arbitrary rings, which is why it's left out. Maybe we should just introduce notation for the cup product, like x ⌣[ R ] y

Comment (thomas-lamiaux) [reply Cubical/Cohomology/EilenbergMacLane/Rings/Z2-properties.agda:66]: I don't know, I think G should have been put explicitly everytime.
I needed it all the time or there is unification issues.
Here, I have just put it like this in a file just for Z/2.

Unless, we plan to compute more cohomology ring with a different ring, I don't think we should bother modifying anything else

File: Cubical/Cohomology/EilenbergMacLane/Rings/RP2wedgeS1.agda [hunk 2]
```diff
@@ -0,0 +1,581 @@
...
+     base-neutral-eq : _
+     base-neutral-eq (zero        ∷ zero              ∷ []) = cong (base 0) (pres1 ϕ₀str) ∙ base-neutral _
+     base-neutral-eq (zero        ∷ one               ∷ []) = cong (base 1) (pres1 ϕ₁str) ∙ base-neutral _
+     base-neutral-eq (zero        ∷ two               ∷ []) = cong (base 2) (pres1 ϕ₂str) ∙ base-neutral _
+     base-neutral-eq (zero        ∷ suc (suc (suc m)) ∷ []) = refl
+     base-neutral-eq (one         ∷ zero              ∷ []) = cong (base 1) (pres1 ϕ₁str) ∙ base-neutral _
+     base-neutral-eq (one         ∷ suc m             ∷ []) = refl
+     base-neutral-eq (suc (suc n) ∷ m                 ∷ []) = refl
+
+     base-add-eq : _
```
Comment (mortberg) [inline Cubical/Cohomology/EilenbergMacLane/Rings/RP2wedgeS1.agda:250]: Long lines

File: Cubical/Cohomology/EilenbergMacLane/Rings/RP2wedgeS1.agda [hunk 3]
```diff
@@ -0,0 +1,581 @@
...
+    pres·-int zero one a zero zero b = cong (baseH* 1) (sym (ϕₙ⌣ϕₘ-notTrivial
+                                       ϕ₁left ϕ₁leftStr ϕ₀ ϕ₀str ϕ₁left ϕ₁leftStr
+                                       ((cong (λ X → ϕ₁left 1ℤ/2 ⌣ℤ/2 X) ϕ₀-pres1)
+                                         ∙ ⌣-1ₕ {G'' = ℤ/2Ring} _ _
+                                         ∙ cong (λ X → subst (λ Y → coHom Y ℤ/2 RP²⋁S¹) X (ϕ₁ (1ℤ/2 , 0ℤ/2)))
+                                                (isSetℕ _ _ (+'-comm zero 1) (refl {x = 1}))
+                                         ∙ substRefl _)
+                                       _ _))
+      -- case α²
+    pres·-int zero one a zero one b = cong (baseH* 2)
```
Comment (mortberg) [inline Cubical/Cohomology/EilenbergMacLane/Rings/RP2wedgeS1.agda:329]: long lines around here

File: Cubical/Cohomology/EilenbergMacLane/Rings/RP2wedgeS1.agda [hunk 4]
```diff
@@ -0,0 +1,581 @@
...
+    ℤ/2[x,y]/<y³,xy,x²>→H*-RP²⋁S¹-pres0 = refl
+
+    ℤ/2[x,y]/<y³,xy,x²>→H*-RP²⋁S¹-pres+ : (x y : ℤ/2[x,y]/<y³,xy,x²>) →
+                                             ℤ/2[x,y]/<y³,xy,x²>→H*-RP²⋁S¹ ( x +Pℤ/2I y)
+                                          ≡ ℤ/2[x,y]/<y³,xy,x²>→H*-RP²⋁S¹ x +H* ℤ/2[x,y]/<y³,xy,x²>→H*-RP²⋁S¹ y
+    ℤ/2[x,y]/<y³,xy,x²>→H*-RP²⋁S¹-pres+ x y = IsRingHom.pres+ (snd ℤ/2[X,Y]/<Y³,XY,X²>→H*R-RP²⋁S¹) x y
+
+
+  -----------------------------------------------------------------------------
+  -- Converse Sens on H* → ℤ/2[X,Y]
```
Comment (mortberg) [inline Cubical/Cohomology/EilenbergMacLane/Rings/RP2wedgeS1.agda:444]:
```suggestion
  -- Converse direction H* → ℤ/2[X,Y]
```

File: Cubical/Cohomology/EilenbergMacLane/Rings/RP2wedgeS1.agda [hunk 5]
```diff
@@ -0,0 +1,581 @@
...
+    e-sect : (x : H*ℤ/2 RP²⋁S¹) → ℤ/2[x,y]/<y³,xy,x²>→H*-RP²⋁S¹ (H*-RP²⋁S¹→ℤ/2[x,y]/<y³,xy,x²> x) ≡ x
+    e-sect = DS-Ind-Prop.f _ _ _ _ (λ _ → isSetH* _ _)
+             refl
+             e-sect-base
+             λ {U V} ind-U ind-V → cong₂ _+H*_ ind-U ind-V
+
+  -----------------------------------------------------------------------------
+  -- Retraction
+
+    e-retr-base : (v : Vec ℕ 2) → (a : fst ℤ/2) →
```
Comment (mortberg) [inline Cubical/Cohomology/EilenbergMacLane/Rings/RP2wedgeS1.agda:520]: long lines in this proof

File: Cubical/Cohomology/EilenbergMacLane/Rings/RP2wedgeS1.agda [hunk 6]
```diff
@@ -0,0 +1,581 @@
...
+
+    e-retr : (x : ℤ/2[x,y]/<y³,xy,x²>) → H*-RP²⋁S¹→ℤ/2[x,y]/<y³,xy,x²> (ℤ/2[x,y]/<y³,xy,x²>→H*-RP²⋁S¹ x) ≡ x
+    e-retr = SQ.elimProp (λ _ → isSetPℤ/2I _ _)
+             (DS-Ind-Prop.f _ _ _ _ (λ _ → isSetPℤ/2I _ _)
+             refl
+             e-retr-base
+             λ {U V} ind-U ind-V → cong₂ _+Pℤ/2I_ ind-U ind-V)
+
+  -----------------------------------------------------------------------------
+  -- Computation of the Cohomology Ring
```
Comment (mortberg) [inline Cubical/Cohomology/EilenbergMacLane/Rings/RP2wedgeS1.agda:562]: Why is all this commented?

File: Cubical/Cohomology/EilenbergMacLane/Rings/Z2-properties.agda (hunk 1 above)
Comment (mortberg) [reply Cubical/Cohomology/EilenbergMacLane/Rings/Z2-properties.agda:66]: I like the `x ⌣[ R ] y` suggestions

Comment (mortberg) [reply Cubical/Cohomology/EilenbergMacLane/Rings/Z2-properties.agda:66]: Also, I wonder if the unification issues could easily be fixed using instance arguments. Seems much nicer than having explicit arguments

**Review by mortberg — CHANGES_REQUESTED** [review]
some comments

File: Cubical/Cohomology/EilenbergMacLane/Rings/Z2-properties.agda (hunk 1 above)
Comment (thomas-lamiaux) [reply Cubical/Cohomology/EilenbergMacLane/Rings/Z2-properties.agda:66]: I don't understand what you want me to do ? 
Shouldn't this be done in Axel file rand reverberate everywhere ?

File: Cubical/Cohomology/EilenbergMacLane/Rings/RP2wedgeS1.agda (hunk 6 above)
Comment (thomas-lamiaux) [reply Cubical/Cohomology/EilenbergMacLane/Rings/RP2wedgeS1.agda:562]: Waiting for @aljungstrom to prove that, I think he branched the PR

File: Cubical/Cohomology/EilenbergMacLane/Rings/Z2-properties.agda (hunk 1 above)
Comment (mortberg) [inline Cubical/Cohomology/EilenbergMacLane/Rings/Z2-properties.agda:66]:
```suggestion
  -- This notation is needed as G'' as Agda can't figure it out. TODO: add a x ⌣[ R ] y notation?
```

**Review by mortberg — APPROVED** [review]

**Pull request merged**